haystacker-server/src/main/kotlin/net/sr89/haystacker/server/handlers/GetBackgroundTaskProgressHandler.kt

```diff
@@ -13,13 +14,23 @@ import org.http4k.core.with
 import java.util.UUID
 
 class GetBackgroundTaskProgressHandler(private val taskManager: BackgroundTaskManager): HttpHandler {
+
+    private fun parseTaskId(taskId: String): TaskId {
+        try {
+            val parsedTaskId = TaskId(UUID.fromString(taskId))
+            return parsedTaskId
+        } catch (iax: IllegalArgumentException) {
+            throw InvalidTaskIdException(taskId)
+        }
+    }
+
     override fun invoke(request: Request): Response {
         val tid: String = taskId(request)
 
-        val status = taskManager.status(TaskId(UUID.fromString(tid)))
+        val status = taskManager.status(parseTaskId(tid))
```

Comment:
If we make `parseTaskId()` part of `TaskId` and rename it to `fromString()`, this would become: `val status = taskManager.status(TaskId.fromString(tid))`